tools/static-assets/server/boot.js

```diff
@@ -390,23 +390,26 @@ var loadServerBundles = Profile("Load server bundles", function () {
     var scriptPath =
       parsedSourceMaps[absoluteFilePath] ? absoluteFilePath : fileInfoOSPath;
 
-    var script = new (require('vm').Script)(wrapped, {
+    var func = require('vm').runInThisContext(wrapped, {
       filename: scriptPath,
       displayErrors: true
     });
 
-    var func = script.runInThisContext();
-
     var args = [Npm, Assets];
 
     specialKeys.forEach(function (key) {
       args.push(specialArgs[key]);
     });
 
-    infos.push({
-      fn: Profile(fileInfo.path, func),
-      args
-    });
+    if (meteorDebugFuture) {
+      infos.push({
+        fn: Profile(fileInfo.path, func),
+        args
+      });
+    } else {
+      // Allows us to use code-coverage if the debugger is not enabled
+      Profile(fileInfo.path, func).apply(global, args);
```

Comment:
In cases when this code executes, I think you need to avoid calling the `func` again after `maybeWaitForDebuggerToAttach` below.

Reply:
The `if` branch adds the `func` function to an array wherefrom it's executed later (after `maybeWaitForDebuggerToAttach `), but the `else` branch, which executes the `func` function immediately, does not add it to the array. By this trick, the method is not called twice here.